### PR TITLE
Made functions: directing vector in 3d, a point on the line and the distance from a point to a straight

### PR DESCRIPTION
**Pull request by efedra**
directing vector in 3d and a point on the line -supporting function

### CODE REVIEW

File: lib/geometry.rb [hunk 1]
```diff
@@ -71,11 +79,93 @@ def minimal_convex_hull_2d(points)
         hull.push(up[j])
       end
       (1..(down.size - 2)).reverse_each do |j|
+        puts j
```
Comment (Wolwer1nE) [inline lib/geometry.rb]: Please, remove puts from release version.

File: lib/geometry.rb [hunk 2]
```diff
@@ -71,11 +79,93 @@ def minimal_convex_hull_2d(points)
...
+    def directing_vector3d(c)
+      c=c.gsub(' ','')
+      c1=c.insert(c.length,'=')
+      res=Array.new()
+      if c1.include?('x')
+        before=c1.index('/')+1
+        after=c1.index('=')
+        res[0]=c1.slice(before..after).gsub('=','').sub('*','').gsub('(','').gsub(')','').to_f
+        c1=c1.slice(after,c1.length).sub('=','')
+      else res[0]=0.0 end
```
Comment (Wolwer1nE) [inline lib/geometry.rb:108]: Place end on next line
if 
else
end

File: lib/geometry.rb [hunk 3]
```diff
@@ -71,11 +79,93 @@ def minimal_convex_hull_2d(points)
...
+      else m[2]=0.0 end
+      return m
+    end
+
+    ##
+    # Calculates the distance from a point given by a Point3d structure
+    # to a straight line given by a canonical equation.
+    # Example, (x-0) / 26 = (y + 300) / * (- 15) = (z-200) / 51
+    #
+    #Important: mandatory order of variables: x, y, z
```
Comment (Wolwer1nE) [inline lib/geometry.rb:163]: Nice comments, thank you.

File: test/geometry_test.rb [hunk 4]
```diff
@@ -44,6 +44,32 @@ def test_distance_point_to_point3d
                     distance_point_to_point3d(Point3d.new(-222,-333,-444),Point3d.new(-2,-5,-6)),0.0001)
   end
 
+  def test_directing_vector3d
+    assert_equal([2.0, 1.0, 2.0],directing_vector3d('(x-3)/2=(y-1)/1=(z+1)/2'))
```
Comment (Wolwer1nE) [inline test/geometry_test.rb:49]: Split these asserts in separate tests

File: lib/geometry.rb (hunk 2 above)
Comment (efedra) [reply lib/geometry.rb]: @Wolwer1nE corrected comments

**Comment by efedra** [comment]
@Wolwer1nE  
corrected comments

**Pull request merged**